### PR TITLE
Push to docs-itb pages if pr branch starts with migrate. or itb.

### PR DESCRIPTION
**Pull request by brianhlin**

### CODE REVIEW

**Comment by bbockelm** [comment]
Talking to BrianL -- this one isn't ready to review yet, has a few more TODOs before it's ready to go in.

**Comment by bbockelm** [comment]
@brianhlin - this one is getting stale.  What's the latest thinking here?

**Comment by brianhlin** [comment]
This had to be put on hold for November release dev work. I should be able to complete this after the dust settles for the the dev freeze today.

**Comment by brianhlin** [comment]
Ok, this is ready for a re-review. 

- [Build from the PR](https://travis-ci.org/opensciencegrid/docs/builds/295512534) only runs the verification (`mkdocs build` and `linkchecker`
- [Build from a branch beginning with itb.*](https://travis-ci.org/opensciencegrid/docs/builds/295511453) runs both the verification and deployment to the itb

A few things:

1. All branches run `mkdocs build` and `linkchecker` before deployment, meaning that transient link failures can prevent deployment (I can change this)
1. ITB documentation is titled differently ([see top left](https://opensciencegrid.github.io/docs-itb/)) but I haven't thought about how to add a warning header/change the readthedocs theme color/whatever to differentiate it from production docs. I also haven't figured out how to modify `robots.txt` so that it isn't crawled
1. `doc-ci-scripts`'s `README.md` needs updated instructions

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,8 @@
...
-- pip install mkdocs
-- pip install MarkdownHighlight
-- pip install pygments
-- pip install linkchecker
-- pip install requests==2.9.2 # https://github.com/wummel/linkchecker/issues/649
+install: pip install -r ci/pip-requirements.txt
 script:
-- if [ "${TRAVIS_PULL_REQUEST}" == "false" ] && [ "${TRAVIS_BRANCH}" == "master" ]; then openssl aes-256-cbc -K $encrypted_75e9e0f098a4_key -iv $encrypted_75e9e0f098a4_iv -in deploy-key.enc -out deploy-key -d && ./travis/run_on_non_pull_requests.sh; else ./travis/run_on_pull_requests.sh; fi
+- ./ci/verify
+- if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then ./ci/deploy; fi
```
Comment (bbockelm) [inline .travis.yml]: Hm.  This seems brittle.

Can you put the `${TRAVIS_PULL_REQUEST}` logic in `./ci/deploy`?  The less logic, the better.

**Comment by brianhlin** [comment]
Yup, Travis performs the submodule checkout. Comments should be addressed now

- Branch deployment [here](https://travis-ci.org/opensciencegrid/docs/builds/298791267?utm_source=github_status&utm_medium=notification)
- PR verification [here](https://travis-ci.org/opensciencegrid/docs/builds/298791265?utm_source=github_status&utm_medium=notification)

**Pull request merged**